File: motor3d 2.0/solucionsistema.vb
Namespace Algebra
    Public Enum TipoSolucionSistema
        SistemaCompatibleDeterminado
        SistemaCompatibleIndeterminado
        SistemaIncompatible
    End Enum

    Public Class SolucionSistema
        Inherits ObjetoAlgebraico

        Private mSolucionCompatibleDeterminado() As Double
        Private mTipoSolucion As TipoSolucionSistema

        Public ReadOnly Property TipoSolucion() As TipoSolucionSistema
            Get
                Return mTipoSolucion
            End Get
        End Property

        Public ReadOnly Property ValorSolucion() As Double()
            Get
                Return mSolucionCompatibleDeterminado
            End Get
        End Property

        Public Sub New(ByVal ParamArray ValSolucion() As Double)
            mTipoSolucion = TipoSolucionSistema.SistemaCompatibleDeterminado
            mSolucionCompatibleDeterminado = ValSolucion
        End Sub

        Public Sub New(ByVal ValTipoSolucion As TipoSolucionSistema)
            mTipoSolucion = ValTipoSolucion
        End Sub

        Public Overrides Function ToString() As String
            Dim Retorno As String = ""

            If mTipoSolucion = TipoSolucionSistema.SistemaCompatibleDeterminado Then
                Retorno = "{Sistema compatible determinado.("
                For i As Integer = 0 To mSolucionCompatibleDeterminado.GetUpperBound(0)
                    If i < mSolucionCompatibleDeterminado.GetUpperBound(0) Then
                        Retorno &= mSolucionCompatibleDeterminado(i).ToString & ","
                    Else
                        Retorno &= mSolucionCompatibleDeterminado(i).ToString & ")}"
                    End If
                Next

                Return Retorno
            Else
                Return "{" & mTipoSolucion.ToString & "}"
            End If
        End Function
    End Class
End Namespace

File: motor3d 2.0.ConvertedToC#/SolucionSistema.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Linq;
using System.Xml.Linq;
namespace Motor3D.Algebra
{
	public enum TipoSolucionSistema
	{
		SistemaCompatibleDeterminado,
		SistemaCompatibleIndeterminado,
		SistemaIncompatible
	}

	public class SolucionSistema : ObjetoAlgebraico
	{

		private double[] mSolucionCompatibleDeterminado;

		private TipoSolucionSistema mTipoSolucion;
		public TipoSolucionSistema TipoSolucion {
			get { return mTipoSolucion; }
		}

		public double[] ValorSolucion {
			get { return mSolucionCompatibleDeterminado; }
		}

		public SolucionSistema(params double[] ValSolucion)
		{
			mTipoSolucion = TipoSolucionSistema.SistemaCompatibleDeterminado;
			mSolucionCompatibleDeterminado = ValSolucion;
		}

		public SolucionSistema(TipoSolucionSistema ValTipoSolucion)
		{
			mTipoSolucion = ValTipoSolucion;
		}

		public override string ToString()
		{
			string Retorno = "";

			if (mTipoSolucion == TipoSolucionSistema.SistemaCompatibleDeterminado) {
				Retorno = "{Sistema compatible determinado.(";
				for (int i = 0; i <= mSolucionCompatibleDeterminado.GetUpperBound(0); i++) {
					if (i < mSolucionCompatibleDeterminado.GetUpperBound(0)) {
						Retorno += mSolucionCompatibleDeterminado[i].ToString() + ",";
					} else {
						Retorno += mSolucionCompatibleDeterminado[i].ToString() + ")}";
					}
				}

				return Retorno;
			} else {
				return "{" + mTipoSolucion.ToString() + "}";
			}
		}
	}
}
